src/azure-cli-core/azure/cli/core/breaking_change.py

```diff
@@ -0,0 +1,596 @@
+            return self.cmd[3:].strip()
+        return self.cmd
+
+    def is_command_group(self, cli_ctx):
+        return self.command_name in cli_ctx.invocation.commands_loader.command_group_table
+
+    def to_tag(self, cli_ctx, **kwargs):
+        if self.args:
+            object_type = 'argument'
+        elif self.is_command_group(cli_ctx):
+            object_type = 'command group'
+        else:
+            object_type = 'command'
+        tag_kwargs = {
+            'object_type': object_type,
+            'target': self.target,
+            'target_version': self.target_version,
+            'message_func': lambda _: self.message,
+        }
+        tag_kwargs.update(kwargs)
+        return UpcomingBreakingChangeTag(cli_ctx, **tag_kwargs)
+
+    def register(self, cli_ctx):
+        def register_option_deprecate(option_name, arguments):
+            for _, argument in arguments.items():
+                if argument.options_list and len(argument.options_list) > 1:
+                    for idx, option in enumerate(argument.options_list):
+                        if isinstance(option, str) and option_name == option and isinstance(self, AzCLIDeprecate):
+                            argument.options_list[idx] = self.to_tag(cli_ctx, object_type='option')
+                            argument.options_list[idx].target = option
+                            argument.action = _argument_breaking_change_action(cli_ctx, argument.options_list[idx],
+                                                                               argument.options['action'])
+                            return True
+            return False
+
+        def appended_status_tag(old_status_tag, new_status_tag):
+            if isinstance(old_status_tag, (Deprecated, UpcomingBreakingChangeTag)):
+                return MergedStatusTag(cli_ctx, old_status_tag, new_status_tag)
+            if isinstance(old_status_tag, MergedStatusTag):
+                old_status_tag.merge(new_status_tag)
+                return old_status_tag
+            return new_status_tag
+
+        def register_to_direct_sub_cg_or_command(cg_name, status_tag):
+            for key, command_group in cli_ctx.invocation.commands_loader.command_group_table.items():
+                if key.rsplit(maxsplit=1)[0] == cg_name:
+                    from azure.cli.core.commands import AzCommandGroup
+                    if isinstance(command_group, AzCommandGroup):
+                        command_group.group_kwargs['deprecate_info'] = \
+                            appended_status_tag(command_group.group_kwargs.get('deprecate_info'), status_tag)
+                    else:
+                        register_to_direct_sub_cg_or_command(key, status_tag)
+            for key, command in cli_ctx.invocation.commands_loader.command_table.items():
+                if key.rsplit(maxsplit=1)[0] == cg_name:
+                    command.deprecate_info = appended_status_tag(command.deprecate_info, self.to_tag(cli_ctx))
```

Comment:
move them as self functions instead of closure functions.

Reply:
Done